server/api/users.js

```diff
@@ -81,15 +81,21 @@ function setSessionCookie(req, res, token) {
   res.cookie('__session', JSON.stringify(cookiePayload), AUTH_COOKIE_OPTION);
 }
 
-function setAuthCookies(req, res, { user, wallet }) {
+async function setAuthCookies(req, res, { user, wallet }) {
   const payload = {
     user,
     wallet,
     permissions: ['read', 'write', 'like'],
   };
-  const token = jwtSign(payload);
+  const { token, jwtid } = jwtSign(payload);
   res.cookie('likecoin_auth', token, AUTH_COOKIE_OPTION);
   setSessionCookie(req, res, token);
+  await dbRef.doc(user).collection('session').doc(jwtid).create({
+    lastAccessedUserAgent: req.headers['user-agent'] || 'unknown',
+    lastAccessedIP: req ? (req.headers['x-real-ip'] || req.ip) : undefined,
```

Comment:
So it is possible to have falsy value `req`? Then `setSessionCookie` should care about this?

Reply:
actually not possible here, copy and paste programming fault..